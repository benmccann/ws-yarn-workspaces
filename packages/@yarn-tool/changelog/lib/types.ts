/**
 * Created by user on 2020/6/15.
 */

import { ITSRequireAtLeastOne } from 'ts-type';

export type IChangelogPreset = string | "conventional-changelog-angular" | "@bluelovers/conventional-changelog-bluelovers";
export type IType = "independent" | "root" | string

export interface IOptionsBaseCore
{
	changelogPreset?: IChangelogPreset,
	rootPath?: string,
}

export interface IOptionsBase extends IOptionsBaseCore
{
	tagPrefix?: string | 'v',
}

export type IOptionsInput = IOptionsBase & ITSRequireAtLeastOne<IOptionsBaseCore>;

export type IOptionsUpdateChangelog = IOptionsInput & {
	version?: string,
}

export type IOptionsRecommendVersion = IOptionsInput & {
	prereleaseId?: string,
}

export type IOptionsWithType<T extends Record<any, any>> = T & {
	type?: IType,
}

export interface IReturnTypeUpdateChangelog
{
	logPath: string;
	newEntry: string;
	version: string;
}
