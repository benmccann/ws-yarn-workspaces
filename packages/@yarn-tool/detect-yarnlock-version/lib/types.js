"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumDetectYarnLock = void 0;
var EnumDetectYarnLock;
(function (EnumDetectYarnLock) {
    EnumDetectYarnLock[EnumDetectYarnLock["v1"] = 1] = "v1";
    /**
     * @deprecated do not use this if u want check is version is berry
     * @type {EnumDetectYarnLock.v2}
     */
    EnumDetectYarnLock[EnumDetectYarnLock["v2"] = 2] = "v2";
    EnumDetectYarnLock[EnumDetectYarnLock["berry"] = 2] = "berry";
    EnumDetectYarnLock[EnumDetectYarnLock["unknown"] = 0] = "unknown";
})(EnumDetectYarnLock = exports.EnumDetectYarnLock || (exports.EnumDetectYarnLock = {}));
//# sourceMappingURL=types.js.map