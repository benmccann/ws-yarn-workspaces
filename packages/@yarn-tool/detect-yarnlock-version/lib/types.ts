
export enum EnumDetectYarnLock
{
	v1 = 1,
	v2 = 2,
	berry = 2,
	unknown = 0,
}
