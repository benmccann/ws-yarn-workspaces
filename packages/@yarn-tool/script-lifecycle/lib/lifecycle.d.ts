export declare const lifecycleMap: {
    install: {
        name: "install";
        ignoreSelf: boolean;
        before: string[];
        after: string[];
    };
    pack: {
        name: "pack";
        ignoreSelf: boolean;
        before: string[];
        after: string[];
    };
    publish: {
        name: "publish";
        ignoreSelf: boolean;
        before: string[];
        after: string[];
    };
};
export default lifecycleMap;
