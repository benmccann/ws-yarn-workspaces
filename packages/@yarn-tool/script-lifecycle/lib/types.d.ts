/**
 * Created by user on 2020/4/9.
 */
import lifecycleMap from "./lifecycle";
export interface ILifecycleEntry<K extends string = string> {
    name: K;
    ignoreSelf?: boolean;
    before: string[];
    after: string[];
}
export declare type ILifecycleMapKeys = keyof typeof lifecycleMap;
export declare type ILifecycleMapEntry<K extends string = string | ILifecycleMapKeys> = K extends ILifecycleMapKeys ? (typeof lifecycleMap)[K] : ILifecycleEntry<K>;
export declare type ILifecycleMap<K extends string = string> = typeof lifecycleMap & Record<K, ILifecycleEntry<K>>;
