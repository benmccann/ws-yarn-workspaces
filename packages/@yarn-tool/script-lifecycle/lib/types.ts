/**
 * Created by user on 2020/4/9.
 */

import lifecycleMap from "./lifecycle";

export interface ILifecycleEntry<K extends string = string>
{
	name: K;
	ignoreSelf?: boolean;
	before: string[];
	after: string[];
}

export type ILifecycleMapKeys = keyof typeof lifecycleMap;

export type ILifecycleMapEntry<K extends string = string | ILifecycleMapKeys> = K extends ILifecycleMapKeys ? (typeof lifecycleMap)[K] : ILifecycleEntry<K>

export type ILifecycleMap<K extends string = string> = typeof lifecycleMap & Record<K, ILifecycleEntry<K>>;
