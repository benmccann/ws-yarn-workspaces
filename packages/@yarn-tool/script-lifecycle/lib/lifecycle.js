"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.lifecycleMap = void 0;
const tslib_1 = require("tslib");
const install_1 = tslib_1.__importDefault(require("./lifecycle/install"));
const pack_1 = tslib_1.__importDefault(require("./lifecycle/pack"));
const publish_1 = tslib_1.__importDefault(require("./lifecycle/publish"));
exports.lifecycleMap = {
    install: install_1.default,
    pack: pack_1.default,
    publish: publish_1.default
};
exports.default = exports.lifecycleMap;
//# sourceMappingURL=lifecycle.js.map