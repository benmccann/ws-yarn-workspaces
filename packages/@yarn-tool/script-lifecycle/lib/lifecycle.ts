
import install from './lifecycle/install'
import pack from './lifecycle/pack'
import publish from './lifecycle/publish'

export const lifecycleMap = {
	install,
	pack,
	publish
}
export default lifecycleMap
