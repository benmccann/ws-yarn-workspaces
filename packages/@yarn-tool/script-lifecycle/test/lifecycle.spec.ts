/**
 * Created by user on 2020/4/9.
 */

import { getLifecycle, entryToList } from '../index';
import { ILifecycleEntry } from '../lib/types';

describe('should lifecycle any script name', function ()
{
	[
		'yy',
		'pretitter',
	].forEach(name => {
		it(name, function ()
		{
			let actual = getLifecycle(name);

			expect(actual).toMatchObject({
				name,
				before: [
					`pre${name}`
				],
				after: [
					`post${name}`
				],
			})

			expect(entryToList(actual)).toContainEqual(name);

			expect(actual.ignoreSelf).not.toBeTruthy();
			expect(actual).toMatchSnapshot();
			expect(entryToList(actual)).toMatchSnapshot();
		});
	});
});

it('install', function ()
{
	let name = 'install';
	let actual = getLifecycle(name);

	expect(actual).toMatchObject({
		name,
		before: [
			"preinstall",
		],
		after: [
			"postinstall",
			"prepublish",
			"prepare",
			"preshrinkwrap",
			"shrinkwrap",
			"postshrinkwrap",
		],
	})

	expect(entryToList(actual)).toContainEqual(name);

	expect(actual).toMatchSnapshot();
	expect(entryToList(actual)).toMatchSnapshot();
});

it('pack', function ()
{
	let name = 'pack';
	let actual = getLifecycle(name);

	expect(actual).toMatchObject({
		name,
		ignoreSelf: true,
		before: [
			"prepublish",
			"prepare",
			"prepack",
		],
		after: [
			"postpack"
		],
	})

	expect(actual).toMatchSnapshot();
	expect(entryToList(actual)).not.toContainEqual(name);
	expect(entryToList(actual)).toMatchSnapshot();

	let actual2 = entryToList(actual, true)

	expect(actual2).toContainEqual(name);
	expect(actual2).toMatchSnapshot();

});

it('publish', function ()
{
	let name = 'publish';
	let actual = getLifecycle(name);

	expect(actual).toMatchObject({
		name,
		before: [
			"prepublish",
			"prepare",
			"prepublishOnly",
			"prepack",
			"postpack",
		],
		after: [
			"postpublish"
		],
	})

	expect(entryToList(actual)).toContainEqual(name);

	expect(actual).toMatchSnapshot();
	expect(entryToList(actual)).toMatchSnapshot();
});
