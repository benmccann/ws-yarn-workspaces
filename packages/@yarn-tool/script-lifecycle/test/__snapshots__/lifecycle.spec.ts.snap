// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`install 1`] = `
Object {
  "after": Array [
    "postinstall",
    "prepublish",
    "prepare",
    "preshrinkwrap",
    "shrinkwrap",
    "postshrinkwrap",
  ],
  "before": Array [
    "preinstall",
  ],
  "ignoreSelf": false,
  "name": "install",
}
`;

exports[`install 2`] = `
Array [
  "preinstall",
  "install",
  "postinstall",
  "prepublish",
  "prepare",
  "preshrinkwrap",
  "shrinkwrap",
  "postshrinkwrap",
]
`;

exports[`pack 1`] = `
Object {
  "after": Array [
    "postpack",
  ],
  "before": Array [
    "prepublish",
    "prepare",
    "prepack",
  ],
  "ignoreSelf": true,
  "name": "pack",
}
`;

exports[`pack 2`] = `
Array [
  "prepublish",
  "prepare",
  "prepack",
  "postpack",
]
`;

exports[`pack 3`] = `
Array [
  "prepublish",
  "prepare",
  "prepack",
  "pack",
  "postpack",
]
`;

exports[`publish 1`] = `
Object {
  "after": Array [
    "postpublish",
  ],
  "before": Array [
    "prepublish",
    "prepare",
    "prepublishOnly",
    "prepack",
    "postpack",
  ],
  "ignoreSelf": false,
  "name": "publish",
}
`;

exports[`publish 2`] = `
Array [
  "prepublish",
  "prepare",
  "prepublishOnly",
  "prepack",
  "postpack",
  "publish",
  "postpublish",
]
`;

exports[`should lifecycle any script name pretitter 1`] = `
Object {
  "after": Array [
    "postpretitter",
  ],
  "before": Array [
    "prepretitter",
  ],
  "ignoreSelf": false,
  "name": "pretitter",
}
`;

exports[`should lifecycle any script name pretitter 2`] = `
Array [
  "prepretitter",
  "pretitter",
  "postpretitter",
]
`;

exports[`should lifecycle any script name yy 1`] = `
Object {
  "after": Array [
    "postyy",
  ],
  "before": Array [
    "preyy",
  ],
  "ignoreSelf": false,
  "name": "yy",
}
`;

exports[`should lifecycle any script name yy 2`] = `
Array [
  "preyy",
  "yy",
  "postyy",
]
`;
